src/components/item/sharing/shortLink/ShortLinksRenderer.tsx

```diff
@@ -44,73 +43,51 @@ const ShortLinksRenderer = ({
 }: Props): JSX.Element => {
   const { data: apiLinks, isLoading } = useShortLinksItem(itemId);
   const { data: publishedEntry } = useItemPublishedInformation({ itemId });
-  const [shortLinks, setShortLinks] = useState<ShortLinkType[]>([]);
   const [modalOpen, setOpen] = useState(false);
   const [isNew, setIsNew] = useState(false);
   const [initialAlias, setInitAlias] = useState<string>('');
   const [initialPlatform, setInitPlatform] = useState<ShortLinkPlatform>(
     Context.Player,
   );
-  // Lists the available short links platforms for the current item.
-  const [itemPlatforms, setItemPlatforms] = useState<ShortLinkPlatformType[]>([
+
+  // List of available platforms, the order is matter
+  const platforms = [
     ShortLinkPlatformConst.builder,
     ShortLinkPlatformConst.player,
-  ]);
-
-  const addNewAlias = useCallback(
-    (
-      platform: ShortLinkPlatform,
-      alias: string,
-      url: URL,
-      isShorten: boolean,
-    ) => {
-      setShortLinks((prevShortLinks) =>
-        [
-          ...prevShortLinks,
-          {
-            linkId: randomString(SHORT_LINK_ID_MAX_LENGTH),
-            alias,
-            platform,
-            url,
-            isShorten,
-          },
-        ].sort((a, b) => a.platform.localeCompare(b.platform)),
-      );
-    },
-    [setShortLinks],
-  );
+    ShortLinkPlatformConst.library,
+  ];
 
-  // Append the library to the short link's platform if item is published
-  useEffect(() => {
-    if (publishedEntry) {
-      setItemPlatforms((currState) => [
-        ...currState,
-        ShortLinkPlatformConst.library,
-      ]);
-    }
-  }, [publishedEntry]);
-
-  useEffect(() => {
-    setShortLinks([]);
-    apiLinks?.forEach((link) => {
-      const url = appendPathToUrl({
-        baseURL: GRAASP_REDIRECTION_HOST,
-        pathname: link.alias,
-      });
-      addNewAlias(link.platform, link.alias, url, true);
-    });
-
-    itemPlatforms.forEach((platform) => {
-      const shortLinksPlatform = apiLinks?.filter(
-        (shortLink) => shortLink.platform === platform,
+  const shortLinks: ShortLinkType[] = platforms
+    .map<ShortLinkType | undefined>((platform) => {
+      if (!publishedEntry && platform === ShortLinkPlatformConst.library) {
+        return undefined;
+      }
+      const clientHostManager = ClientHostManager.getInstance();
+      const url = clientHostManager.getItemAsURL(platform, itemId);
+
+      const shortLink = {
+        linkId: randomString(SHORT_LINK_ID_MAX_LENGTH),
```

Comment:
Is this only used as a key for react ? You could use the platform for the key I think. Because here it will change every time which is contrary to the purpose of a key in react ...